### ISSUE TITLE
Cannot read property 'onRouteChangeStart' of undefined

### ISSUE THREAD

**Issue by brianlovin** [sergiodxa/next-nprogress#32]
Hey @sergiodxa - once I upgraded to 1.3.0 my app fails to run with this error:

```js
Uncaught TypeError: Cannot read property 'onRouteChangeStart' of undefined
    at b.i (component.js:formatted:1)
    at commitLifeCycles (react-dom.development.js:15255)
    at commitAllLifeCycles (react-dom.development.js:16523)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)
    at commitRoot (react-dom.development.js:16677)
    at completeRoot (react-dom.development.js:18069)
    at performWorkOnRoot (react-dom.development.js:17997)
    at performWork (react-dom.development.js:17901)
    at performSyncWork (react-dom.development.js:17873)
    at requestWork (react-dom.development.js:17761)
    at scheduleWork (react-dom.development.js:17566)
    at scheduleRootUpdate (react-dom.development.js:18240)
    at updateContainerAtExpirationTime (react-dom.development.js:18267)
    at updateContainer (react-dom.development.js:18324)
    at ReactRoot.render (react-dom.development.js:18586)
    at eval (react-dom.development.js:18726)
    at unbatchedUpdates (react-dom.development.js:18124)
    at legacyRenderSubtreeIntoContainer (react-dom.development.js:18722)
    at Object.hydrate (react-dom.development.js:18780)
    at renderReactElement (index.js:334)
    at _callee5$ (index.js:397)
    at tryCatch (runtime.js:62)
    at Generator.invoke [as _invoke] (runtime.js:288)
    at Generator.prototype.(:3000/anonymous function) [as next] (http://localhost:3000/_next/static/runtime/main.js:16427:21)
    at asyncGeneratorStep (asyncToGenerator.js:5)
    at _next (asyncToGenerator.js:27)
    at asyncToGenerator.js:34
    at new Promise (<anonymous>)
    at new F (_export.js:36)
    at asyncToGenerator.js:23
    at _doRender (index.js:457)
    at doRender (index.js:343)
    at _callee2$ (index.js:233)
    at tryCatch (runtime.js:62)
    at Generator.invoke [as _invoke] (runtime.js:288)
    at Generator.prototype.(:3000/anonymous function) [as next] (http://localhost:3000/_next/static/runtime/main.js:16427:21)
    at asyncGeneratorStep (asyncToGenerator.js:5)
    at _next (asyncToGenerator.js:27)
    at asyncToGenerator.js:34
    at new Promise (<anonymous>)
    at new F (_export.js:36)
    at asyncToGenerator.js:23
    at _render (index.js:254)
    at render (index.js:205)
    at _callee$ (index.js:185)
    at tryCatch (runtime.js:62)
    at Generator.invoke [as _invoke] (runtime.js:288)
    at Generator.prototype.(:3000/anonymous function) [as next] (http://localhost:3000/_next/static/runtime/main.js:16427:21)
```

This is in combination with React 16.6.

Here is a branch with repo: https://github.com/brianlovin/brian-lovin-next/tree/react-and-fixes
clone, yarn, and `yarn run dev` should repro this error. It appears the error likely came from https://github.com/sergiodxa/next-nprogress/pull/27/files but it's hard to say.

Curious if you've seen this (I couldn't find anything too similar in past issues). In the meantime I'm going to try and fork + fix a local version of this package.

**Comment by brianlovin** [comment]
In my cloned version I consoled the props coming in to `component.js`:

```
color: "#2299DD"
delayMs: 500
options: undefined
router: undefined
showAfterMs: 300
spinner: true
```

That `undefined` `router` seems to be the culprit. Will keep digging.

**Comment by brianlovin** [comment]
I should additionally note that this is with the higher order component implementation, wrapping my `_app.js` component

**Comment by brianlovin** [comment]
Unfortunately I can't figure out where this is breaking. In my clone I'm able to get the imported `next-nprogress/component` to not break the site, but then it doesn't insert the `#nprogress` dom node

**Comment by brianlovin** [comment]
I ended up removing `next-nprogress` from https://github.com/brianlovin/brian-lovin-next/tree/react-and-fixes so that I can build and deploy the site.

**Comment by sergiodxa** [comment]
I can replicate it on my own site, will check what's happening.

**Comment by denkigai** [comment]
I found that Next.js now handles route change events based on EventEmitter, so you should change 
this part (https://github.com/sergiodxa/next-nprogress/blob/master/src/component.js#L14-L47)
to like below. (I cloned this repo and change it like below and at local it works as expected)

```
componentDidMount() {
    const { options, router, showAfterMs } = this.props

    if (options) {
      NProgress.configure(options)
    }

    router.events.on('routeChangeStart', () => {
      NProgress.start()
    })

    router.events.on('routeChangeComplete', () => {
      NProgress.done()
    })

    router.events.on('routeChangeError', () => {
      NProgress.done()
    })
  }
```
and this is documented [here](https://github.com/zeit/next.js/#router-events).


(However, it seems like Next.js also handles the methods like`router.onRouteChangeStart`  as far as I see the [code here](https://github.com/zeit/next.js/blob/canary/packages/next-server/lib/router/index.js#L49-L63). Is it never called ?)

**Comment by brianlovin** [comment]
Ping @sergiodxa - seems like we have a possible fix here in #36, if possible, could this be published?

**Comment by brianlovin** [comment]
Also confirming that 1.4.0 fixes the problem locally :)
